Classes/URKArchive.mm

```diff
@@ -385,23 +395,75 @@ + (BOOL)urlIsARAR:(NSURL *)fileURL
     return [NSArray arrayWithArray:fileInfos];
 }
 
-- (nullable NSString *)firstVolumePath {
-    return @"";
+- (NSString *)firstVolumePath:(NSString *)filePath {
+    NSURL *fileURL = [self firstVolumeURL:[NSURL fileURLWithPath:filePath]];
+
+    return fileURL.path;
 }
 
-- (nullable NSURL *)firstVolumeURL {
-    NSString *path = [self firstVolumePath];
+- (NSURL *)firstVolumeURL:(NSURL *)fileURL {
```

Comment:
Changed the contents to the URL method because `init` uses URL

Reply:
I have a few thoughts on this.

1. Since we're working with regular expressions in the `firstVolume...` method, it makes sense for it to use a path. Let it be true to itself
2. I think it would be clearer if `firstVolumePath` were a class method that takes a path, since it isn't using any internal state of the class
3. It only needs to be private. We can get rid of the public `firstVolumeURL` and `firstPathURL` instance methods, since all instances are going to represent the first volume. Basically, it will always be true that `firstVolumeURL == fileURL`